### DISCUSSION TITLE
Change menu titles

### DISCUSSION THREAD

**Discussion by teamcrisis** [thuliteio/doks#115] [Q&A] [answered]
I'm new to Hugo and this template. I'm having a bit of trouble I hope someone can help with.

Regarding this: https://github.com/h-enk/doks/blob/master/layouts/partials/sidebar/docs-menu.html#L9

It seems like the menu item titles are based off the title of the page. My page titles can be a bit long, so I'd prefer to specify my own front matter for the menu title.

I called it `menutitle`, but how do I get it to show within the menu? Everything I tried results in errors or it shows the same value for all menu items. I think `range` might be my issue, but I'm not familiar enough with Hugo to get past the problem.

**Comment by teamcrisis** [comment, marked as answer]
Looks like I just need to do this:
```
menu:
  docs:
    parent: "help"
    name: "My Name"
```

**Reply by epogrebnyak** [reply]
I have been thinking about that case too, glad you spotted and provided a solution, thank you!

**Reply by teamcrisis** [reply]
@epogrebnyak I also noticed that `docs-navigation.html` needs some adjustment when making this change.

This is what I did:
```
{{ if or .Prev .Next -}}
	<div class="docs-navigation d-flex justify-content-between">
	<!-- https://www.feliciano.tech/blog/custom-sort-hugo-single-pages/ -->
	{{ $pages := where site.RegularPages "Section" .Section -}}
	{{ with $pages.Next . -}}
		<a href="{{ .Permalink }}">
			<div class="card my-1">
				<div class="card-body py-2">
					{{ if isset .Params.Menu.docs "name" }}
					    &larr; {{ .Params.Menu.docs.name }}
					{{ else }}
					    &larr; {{ .Title }}
					{{ end }}
				</div>
			</div>
		</a>
	{{ end -}}
	{{ with $pages.Prev . -}}
		<a class="ml-auto" href="{{ .Permalink }}">
			<div class="card my-1">
				<div class="card-body py-2">
					{{ if isset .Params.Menu.docs "name" }}
					    {{ .Params.Menu.docs.name }} &rarr;
					{{ else }}
					    {{ .Title }} &rarr;
					{{ end }}
				</div>
			</div>
		</a>
	{{ end -}}
	</div>
{{ end -}}
```

Basically, if the menu name is set, then it will use it, otherwise it falls back to the standard title.

I'm not sure this is the best or most elegant way of handling this, but it's all I can come up with, with my Hugo n00b status.

**Reply by epogrebnyak** [reply]
Do you develop source in a repo? Would be nice to have link, easier to follow the change

**Reply by h-enk** [reply]
> Basically, if the menu name is set, then it will use it, otherwise it falls back to the standard title.

Nice one — I will add this to Doks with slightly different code (just my preference):

```html
{{ if or .Prev .Next -}}
	<div class="docs-navigation d-flex justify-content-between">
	<!-- https://www.feliciano.tech/blog/custom-sort-hugo-single-pages/ -->
	{{ $pages := where site.RegularPages "Section" .Section -}}
	{{ with $pages.Next . -}}
		<a href="{{ .Permalink }}">
			<div class="card my-1">
				<div class="card-body py-2">
					{{with .Params.Menu.docs.name }}
						&larr; {{ . }}
					{{ else }}
						&larr; {{ .Title }}
					{{ end }}
				</div>
			</div>
		</a>
	{{ end -}}
	{{ with $pages.Prev . -}}
		<a class="ml-auto" href="{{ .Permalink }}">
			<div class="card my-1">
				<div class="card-body py-2">
					{{with .Params.Menu.docs.name }}
						{{ . }} &rarr;
					{{ else }}
						{{ .Title }} &rarr;
					{{ end }}
				</div>
			</div>
		</a>
	{{ end -}}
	</div>
{{ end -}}
```
